src/foram3d/foram.ts

```diff
@@ -76,38 +80,66 @@ module Foram3D {
       newChamber.aperture = newAperture;
       newChamber.setAncestor(this.currentChamber);
 
+      this.prevChambers[2] = this.prevChambers[1];
+      this.prevChambers[1] = this.prevChambers[0];
+      this.prevChambers[0] = newChamber;
+
       return newChamber;
     }
 
     private calculateNewCenter(): THREE.Vector3 {
```

Comment:
OK, moim zdaniem jest to algorytm zgodny z tym moim dokumentem, więc jeśli będzie źle, możesz zrzucać na mnie. Jak chodzi o sam kod, proponowałbym:

* zmienić nazwę `deviationSurfaceSpanning` na... coś innego. `secondAperturePlaneSpanningVector`?. I wierzyć w minificatory JS. (swoją drogą, surface to powierzchnia bardziej, plane - płaszczyzna, z tego co wiem)
* zmienić nazwę `phiDeviationAxis` na `aperturePlaneNormal`
* powywalać niektóre rzeczy do pomocniczych metod:
  * `findReferenceLine()`
  * `findAperturePlaneNormal()`
  * `performDeviation(...)`

  Żeby ta metoda była nieco krótsza i kod był czytelniejszy.